pandas/plotting/_core.py

```diff
@@ -788,6 +788,11 @@ def _get_axes_layout(self):
         for ax in axes:
             # check axes coordinates to estimate layout
             points = ax.get_position().get_points()
+            # in IPython inline backend, floats returned by
+            #  `get_points()` have too many ad hoc trailing
+            # digits. Unless rounded these values won't
+            # match in the following set operations
+            points = np.round(points, 3)
```

Comment:
I worry about this causing debugging nightmares further down the road... I'd prefer to solve this local to the problem if possible.

If not, you could perhaps include a keyword to `_get_axes_layout`? I'm not sure how difficult that would be to plumb through.

Reply:
I agree. That's pretty tough to track. 

I made a new commit. The changes are local now. I'm forcing colorbars to take on their relevant axes positions from the parent axes. The merge passes the tests and resolves the xlabel issue (#10611 and #10678) but not with the subplots (#20455).